File: lib/models/categories_model.dart
import 'package:isar/isar.dart';

part 'categories_model.g.dart';

@Collection()
class CategoriesModel {
  Id id = Isar.autoIncrement;

  late String categoryName;
  late int iconCodePoint;

  CategoriesModel({
    required this.categoryName,
    required this.iconCodePoint
  });

}

File: src/models/categories_model.js
import { v4 as uuidv4 } from 'react-native-uuid';

class CategoriesModel {
  constructor({ categoryName, iconCodePoint }) {
    this.id = uuidv4();
    this.categoryName = categoryName;
    this.iconCodePoint = iconCodePoint;
  }
}

export default CategoriesModel;